### PR TITLE
Add libfabric versions 1.15.0 and 1.15.1

### PR DESCRIPTION
**Pull request by teojgo**

### CODE REVIEW

File: var/spack/repos/builtin/packages/libfabric/package.py [hunk 1]
```diff
@@ -45,6 +47,7 @@ class Libfabric(AutotoolsPackage):
                'gni',
                'mlx',
                'mrail',
+               'opx',
```
Comment (haampie) [inline var/spack/repos/builtin/packages/libfabric/package.py]:
```suggestion
               conditional('opx', when='@1.15.0:'),
```

File: var/spack/repos/builtin/packages/libfabric/package.py [hunk 2]
```diff
@@ -98,6 +102,9 @@ class Libfabric(AutotoolsPackage):
     conflicts('@1.9.0', when='platform=darwin',
               msg='This distribution is missing critical files')
 
+    # Omni-Path Express (OPX) libfabric provider was added in 1.15.0
+    conflicts('fabrics=opx', when='@:1.14')
+
```
Comment (haampie) [inline var/spack/repos/builtin/packages/libfabric/package.py]:
```suggestion
```

**Review by haampie — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/libfabric/package.py (hunk 2 above)
Comment (haampie) [reply var/spack/repos/builtin/packages/libfabric/package.py]: and line 107, somehow github doesn't pick it up

File: var/spack/repos/builtin/packages/libfabric/package.py [hunk 3]
```diff
@@ -45,6 +47,8 @@ class Libfabric(AutotoolsPackage):
                'gni',
                'mlx',
                'mrail',
+               conditional('opx', when='@1.15.0:'),
+               'opx',
```
Comment (haampie) [inline var/spack/repos/builtin/packages/libfabric/package.py]:
```suggestion
```

**Review by haampie — APPROVED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

**Review by raghupazl — APPROVED** [review]

**Review by rajachan — APPROVED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

**Comment by haampie** [comment]
the gitlab failure is legit

```python
==> [2022-06-27-17:17:17.041971] Error: TypeError: can only concatenate str (not "_ConditionalVariantValues") to str
/builds/spack/spack/var/spack/repos/builtin/packages/libfabric/package.py:125, in configure_args:
        122            args.append('--with-kdreg=no')
        123
        124        for fabric in self.fabrics:
  >>    125            if 'fabrics=' + fabric in self.spec:
        126                args.append('--enable-{0}=yes'.format(fabric))
        127            else:
        128                args.append('--enable-{0}=no'.format(fabric))
See build log for details:
```

File: var/spack/repos/builtin/packages/libfabric/package.py [hunk 4]
```diff
@@ -17,6 +17,8 @@ class Libfabric(AutotoolsPackage):
     maintainers = ['rajachan']
 
     version('master', branch='master')
+    version('1.15.1', sha256='cafa3005a9dc86064de179b0af4798ad30b46b2f862fe0268db03d13943e10cd')
+    version('1.15.0', sha256='70982c58eadeeb5b1ddb28413fd645e40b206618b56fbb2b18ab1e7f607c9bea')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libfabric/package.py:21]: These have been merged already to `develop`

**Review by alalazo — CHANGES_REQUESTED** [review]

**Comment by teojgo** [comment]
The suggested versions are already added in the develop branch

**Pull request closed without merging**